File: 647-palindromic-substrings/palindromic-substrings.cpp
class Solution {
public:
    int countSubstrings(std::string s) {
        int count = 0;
        int n = s.length();

        for (int i = 0; i < n; ++i) {
            for (int j = i; j < n; ++j) {
                std::string substr = s.substr(i, j - i + 1);
                if (isPalindrome(substr)) {
                    count++;
                }
            }
        }

        return count;
    }

private:
    bool isPalindrome(std::string str) {
        int left = 0, right = str.length() - 1;
        while (left < right) {
            if (str[left] != str[right]) {
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
};

File: 647-palindromic-substrings/palindromic-substrings.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        count = 0
        n = len(s)

        for i in range(n):
            for j in range(i, n):
                substr = s[i:j+1]
                if substr == substr[::-1]: # checking string with the reverse
                    count += 1

        return count
        